streaming/src/test/scala/org/apache/spark/streaming/CheckpointSuite.scala

```diff
@@ -613,7 +613,8 @@ class CheckpointSuite extends TestSuiteBase with DStreamCheckpointTester
     def recordedFiles(ssc: StreamingContext): Seq[Int] = {
       val fileInputDStream =
         ssc.graph.getInputStreams().head.asInstanceOf[FileInputDStream[_, _, _]]
-      val filenames = fileInputDStream.batchTimeToSelectedFiles.values.flatten
+      val filenames = fileInputDStream.batchTimeToSelectedFiles.synchronized
```

Comment:
I'm OK with this, but it would be cleaner to make `fileInputDStream.batchTimeToSelectedFiles` a val and reuse it, and keep the open brace with `synchronized`